### DISCUSSION TITLE
Nextjs Link component & non-public routes

### DISCUSSION THREAD

**Discussion by MacaylaMarvelous81** [deta/space-docs#28] [Q&A]
I think that the Link component from nextjs enables navigating from public routes to non-public routes without authentication, since it loads a json file from _next instead. How do I stop this from happening? do I have to use anchor tags, or list files in _next, use another micro, or something else?
My public_routes look like this:
```yaml
public_routes:
  - /
  - /read
  - /read/*
  - /_next/*
  - /assets/*
```
I can't access /dashboard directly, but I can navigate to it through a Link.

**Comment by sudo-subg** [comment]
@MacaylaMarvelous81 I tried to replicate this, but in my tests everything works as intended, including anchor tags initialized with `Link` components that are handled as they are specified in the `public_routes` like given below:
```yaml
public_routes:
  - /
  - _next/static/*
  - route/*
  - about
```

**Reply by MacaylaMarvelous81** [reply]
I can't get it working with Link, I tried _next/static/* instead of _next/* but the Link still leads to the page without authentication. It even takes me to the page when I use legacyBehavior. It does work if I use a tags instead of Link, but it reloads the page every time which I would like to avoid

**Reply by sudo-subg** [reply]
It would be helpful if you can create a repo with minimal source code base to replicate this

**Comment by MacaylaMarvelous81** [comment]
Alright, I made an example at MacaylaMarvelous81/space-public-link-ex. There's also a builder instance [here](https://splex-1-a7786960.deta.app).
The third link at the top 'Secret' isn't supposed to be able to be accessed with the link.

Sorry about my delay :)
